src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -230,44 +231,45 @@ private void SetEntry(CacheEntry entry)
                 }
             }
 
-            StartScanForExpiredItems(utcNow);
+            StartScanForExpiredItemsIfNeeded(utcNow);
         }
 
         /// <inheritdoc />
         public bool TryGetValue(object key, out object result)
         {
             ValidateCacheKey(key);
-
             CheckDisposed();
 
-            result = null;
             DateTimeOffset utcNow = _options.Clock.UtcNow;
-            bool found = false;
 
             if (_entries.TryGetValue(key, out CacheEntry entry))
             {
                 // Check if expired due to expiration tokens, timers, etc. and if so, remove it.
                 // Allow a stale Replaced value to be returned due to concurrent calls to SetExpired during SetEntry.
-                if (entry.CheckExpired(utcNow) && entry.EvictionReason != EvictionReason.Replaced)
+                if (!(entry.CheckExpired(utcNow) && entry.EvictionReason != EvictionReason.Replaced))
```

Comment:
I'm curious why you flipped the polarity here?

Reply:
I just wanted to make the "hit" code path being the first branch (I expect that most of the calls to the cache should result in finding an element)

Reply:
Could you at least distribute the negation then?
```C#
if (!entry.CheckExpired(utcNow) || entry.EvictionReason == EvictionReason.Replaced)
```